lib/ex_doc/formatter/html/templates.ex

```diff
@@ -84,12 +84,32 @@ defmodule ExDoc.Formatter.HTML.Templates do
     for extra <- extras do
       %{id: id, title: title, group: group, content: content} = extra
 
+      search_data =
+        case extra do
+          %{search_data: search_data} when is_list(search_data) ->
+            Enum.map(search_data, fn search_item ->
+              search_item
+              |> Map.drop([:type, :body])
+              |> Map.put(:labels, [search_item.type])
+            end)
+
+          _ ->
+            nil
+        end
+
       %{
         id: to_string(id),
         title: to_string(title),
         group: to_string(group),
         headers: extract_headers(content)
       }
+      |> then(fn item ->
+        if search_data do
+          Map.put(item, :searchData, search_data)
```

Comment:
Sorry, something go lost in translation. We should not send the search data as is, but the transformed version of the search data. Make it look like `headers` as much as possible. I'd rather transform it on Elixir code than JS code. :D

Reply:
Right 🤦‍♂️

Reply:
This came from an intermediate design when I though I had to support both the child node and section branches with a single object and IIRC they had different fields, but the conditional for "if searchData" solves that.